File: spring-integration/filtering-messages/custom-filter/src/main/java/com/ddubson/Application.java
package com.ddubson;

import com.ddubson.gateways.PrinterGateway;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ImportResource;
import org.springframework.messaging.Message;
import org.springframework.messaging.support.MessageBuilder;

import java.util.stream.Stream;

import static com.ddubson.ANSIColor.ANSI_WHITE;
import static com.ddubson.ANSIColor.ANSI_YELLOW;

@SpringBootApplication
@ImportResource("integration-context.xml")
public class Application implements ApplicationRunner {
    @Autowired
    private PrinterGateway gateway;

    @Override
    public void run(ApplicationArguments args) throws Exception {
        logAdapter().info("[1] Send messages to inputChannel.", ANSI_WHITE);
        Stream.iterate(0, n -> n + 1).limit(10).forEach(i -> {
            Message<?> intMessage = MessageBuilder.withPayload(i).build();
            logAdapter().info("[1] Send message " + i, ANSI_YELLOW);
            this.gateway.print(intMessage);
        });
    }

    @Bean
    public LogAdapter logAdapter() {
        return new CommandLineLogAdapter();
    }

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}

File: spring-integration/routing-messages/payload-type-router/src/main/kotlin/com/ddubson/integration/Application.kt
package com.ddubson.integration

import com.ddubson.ANSIColor
import com.ddubson.LogAdapter
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.messaging.support.MessageBuilder

@SpringBootApplication
class Application(val logAdapter: LogAdapter,
                  val gateway: PrinterGateway) : ApplicationRunner {

    override fun run(args: ApplicationArguments?) {
        logAdapter.info("[1] Send messages to gateway.", ANSIColor.ANSI_WHITE)

        (0..10).forEach { i ->
            val message = MessageBuilder.withPayload("Message " + i).build()
            logAdapter.info("[1.1] Send string message ${message.payload}", ANSIColor.ANSI_YELLOW)
            gateway.print(message)
        }

        (0..10).forEach { i ->
            val message = MessageBuilder.withPayload(i).build()
            logAdapter.info("[1.2] Send int message: ${message.payload}", ANSIColor.ANSI_PURPLE)
            gateway.print(message)
        }
    }

}

fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}
